class-model/src/main/java/org/glassfish/hk2/classmodel/reflect/impl/MethodSignatureVisitorImpl.java

```diff
@@ -62,6 +67,11 @@ public SignatureVisitor visitParameterType() {
         return this;
     }
 
+    @Override
+    public SignatureVisitor visitExceptionType() {
+        return this;
```

Comment:
Shouldn't `exceptionTypes` be updated somewhere, perhaps here? Otherwise - how it gets populated?

Reply:
I agree with @pzygielo. This code doesn't do anything right now.

Reply:
I will review - I was simply cherry-picking from our fork (I didn't make the original change)

Reply:
This is an inherited method from the `SignatureVisitor` class from ASM: [SignatureVisitor#L153](https://gitlab.ow2.org/asm/asm/-/blob/ASM_9_7/asm/src/main/java/org/objectweb/asm/signature/SignatureVisitor.java?ref_type=tags#L153).

There is an identical implementation of it over in the `SignatureVisitorImpl` class here in HK2: [SignatureVisitorImpl#L88](https://github.com/eclipse-ee4j/glassfish-hk2/blob/master/class-model/src/main/java/org/glassfish/hk2/classmodel/reflect/impl/SignatureVisitorImpl.java#L88).

So I assume the original intent was to simply mirror the behaviour of the `SignatureVisitorImpl` class.

Reply:
Could be to follow established pattern, right.
What would be the role of `exceptionTypes`?

Reply:
Would need to spend some time digging into it with a debugger - it's a bit abstract for my eyes to dry-run through.

The interface method from `MethodModel` (and presumably `MethodModelImpl`) gets used over in Payara for MicroProfile OpenAPI processing [here](https://github.com/payara/Payara/blob/payara-server-6.2024.5/appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java#L983) and [here](https://github.com/payara/Payara/blob/payara-server-6.2024.5/appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java#L1287).

Within HK2 itself the important changes within this PR seem to be working on the `MethodModel` within the `ModelClassVisitor` ([here](https://github.com/eclipse-ee4j/glassfish-hk2/pull/1031/files#diff-5c431be86ccad64b09823d78bbf5bb0ac98618fefcca609fe10152b40a527c34R284)). [This code is preceded by reading the method signature](https://github.com/eclipse-ee4j/glassfish-hk2/pull/1031/files#diff-5c431be86ccad64b09823d78bbf5bb0ac98618fefcca609fe10152b40a527c34R272-R274) (which is where this `MethodSignatureVisitorImpl` class comes into play), [during which at least one code path seems to "visit" the exception type](https://gitlab.ow2.org/asm/asm/-/blob/ASM_9_7/asm/src/main/java/org/objectweb/asm/signature/SignatureReader.java?ref_type=tags#L117) (which is where I start getting lost without a debugger).

Reply:
@Pandrex247, list `exceptionTypes` at L39 always stay empty because it never updated.

Reply:
> list `exceptionTypes` at L39 always stay empty because it never updated.

Probably we don't know. `getExceptionTypes` returns original, mutable list so this MIGHT be updated anywhere, out of control of the class.
I understand `getParameters` does the same.

Reply:
> I understand `getParameters` does the same.

But `parameters` collection is filled in the `visitParameterType()` method at L63 of the `MethodSignatureVisitorImpl` class.

Actual exception types processing occurs in the `ModelClassVisitor` class beginnin with L284.

> mutable list so this MIGHT be updated anywhere, out of control of the class.

In this case even out of control HK2.